prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -530,11 +620,99 @@ def canonicalize_dataset_automatically(
             )  # noqa E501
             return canonicalized_dataset
 
+    def get_datasets_of_required_size(
+        self, datasets_info_dict: dict, prompt_spec: PromptSpec
+    ) -> datasets.DatasetDict:
+        """Combine multiple datasets to get the required size.
+
+        Args:
+            dataset_list (list[dict]): A list of dictionaries representing the datasets.
+            prompt_spec (PromptSpec): An object representing the prompt specification.
+
+        Returns:
+            transformed_dataset: The transformed dataset containing the required
+            number of points.
+
+        """
+        curr_datasets_size = 0
+        inputs = []
+        outputs = []
+        dataset_contributions = {}
+        number_of_chosen_datasets = 0
+        while (
+            curr_datasets_size < self.num_points_to_transform
+            and len(datasets_info_dict.keys()) > 0
+            and number_of_chosen_datasets <= self.max_datasets_to_choose
+        ):
+            dataset_name, config_name = self.rerank_datasets(
+                datasets_info_dict, prompt_spec
+            )
+            if dataset_name is None:
+                # If it couldn't find a relevant dataset (even after voting)
+                # stop trying to find more datasets.
+                return None
```

Comment:
but we might have already transformed and created a part of the dataset right? why are we not returning whatever we were able to transform? We can give a warning that says requested X, created Y

Reply:
Sorry I'm not sure I follow - this is for reranking voting (where dataset name returned is None, probably due to hallucinations). I'll make this clearer in comments